File: complejidad/problema1/check_password.py
def check_password_strength(password):
    if len(password) < 8:
        return "La contraseña es muy corta"
    
    #criterios
    has_uppercase = False
    has_lowercase = False
    has_number = False
    has_symbol = False

    for char in password:
        if char.isupper():
            has_uppercase = True
        elif char.islower():
            has_lowercase = True
        elif char.isdigit():
            has_number = True
        elif char in "!@#$%^&*()_+-=,./<>?;:[]{}\|":
            has_symbol = True

    if not has_uppercase:
        return "La contraseña debe tener una mayúscula"
    if not has_lowercase:
        return "La contraseña debe tener una minúscula"
    if not has_number:
        return "La contraseña debe tener un número"
    if not has_symbol:
        return "La contraseña debe tener un símbolo"
    
    return "La contraseña es segura"

if __name__ == "__main__":
    password = input("Ingrese la contraseña: ")
    print(check_password_strength(password))

File: complejidad/problema1/src/checker/CheckPassword.java
package checker;

import java.io.BufferedReader;
import java.io.InputStreamReader;

public class CheckPassword {

	
	public String checkPasswordStrenght(char[] pass) {
		if(pass.length < 8)
			return "La contraseña es muy corta";
		
		boolean hasUppercase = false;
		boolean hasLowercase = false;
		boolean hasNumber = false;
		boolean hasSymbol = false;
		char[] specialChars = "!@#$%^&*()_+-=,./<>?;:[]{}|".toCharArray();
		
		for(char s : pass) {
			if(Character.isUpperCase(s)) {
				hasUppercase = true;
			} else if(!Character.isUpperCase(s)) {
				hasLowercase = true;
			} else if(Character.isDigit(s)) {
				hasNumber = true;
			} else {
				for( int i=0; i<specialChars.length; i++) {
					if(specialChars[i] == s) {
						hasSymbol = true;
						break;
					}
				}
			}
			
			if(!hasUppercase) {
				return "La contraseña debe tener una mayúscula";
			} 
			if(!hasLowercase) {
				return "La contraseña debe tener una minúscula";
			}
			if(!hasNumber) {
				return "La contraseña debe tener un número";
			}
			if(!hasSymbol) {
				return "La contraseña debe tener un símbolo especial como !@#$%^&*()_+-=,./<>?;:[]{}|";
			}
		}
		return "la contraseña es segura";
	}
	
	public static void main(String[] args) {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		try {
			System.out.println("Ingrese la contraseña: ");
			String password = br.readLine();
			char[] pass = password.toCharArray();
			CheckPassword checker =  new CheckPassword();
			String res = checker.checkPasswordStrenght(pass);
			System.out.println(res);
		} catch(Exception e) {
			System.out.println(e.getMessage());
		}
	}
}
